nibabel/viewers.py

```diff
@@ -0,0 +1,473 @@
+""" Utilities for viewing images
+
+Includes version of OrthoSlicer3D code originally written by our own
+Paul Ivanov.
+"""
+from __future__ import division, print_function
+
+import numpy as np
+import weakref
+
+from .optpkg import optional_package
+from .orientations import aff2axcodes, axcodes2ornt
+
+
+class OrthoSlicer3D(object):
+    """Orthogonal-plane slicer.
+
+    OrthoSlicer3d expects 3-dimensional data, and by default it creates a
+    figure with 3 axes, one for each slice orientation.
+
+    Clicking and dragging the mouse in any one axis will select out the
+    corresponding slices in the other two. Scrolling up and
+    down moves the slice up and down in the current axis.
+
+    OrthoSlicer3d also supports 4-dimensional data, where multiple
+    3-dimensional volumes are stacked along the last axis.  For 4-dimensional
+    data the fourth figure axis can be used to control which 3-dimensional
+    volume is displayed.  Alternatively, the - key can be used to decrement the
+    displayed volume and the + or = keys can be used to increment it.
+
+    Example
+    -------
+    >>> import numpy as np
+    >>> a = np.sin(np.linspace(0,np.pi,20))
+    >>> b = np.sin(np.linspace(0,np.pi*5,20))
+    >>> data = np.outer(a,b)[..., np.newaxis]*a
+    >>> OrthoSlicer3D(data).show()  # doctest: +SKIP
+    """
+    # Skip doctest above b/c not all systems have mpl installed
+    def __init__(self, data, affine=None, axes=None, cmap='gray',
+                 pcnt_range=(1., 99.), figsize=(8, 8), title=None):
+        """
+        Parameters
+        ----------
+        data : ndarray
+            The data that will be displayed by the slicer. Should have 3+
+            dimensions.
+        affine : array-like | None
+            Affine transform for the data. This is used to determine
+            how the data should be sliced for plotting into the saggital,
+            coronal, and axial view axes. If None, identity is assumed.
+            The aspect ratio of the data are inferred from the affine
+            transform.
+        axes : tuple of mpl.Axes | None, optional
+            3 or 4 axes instances for the 3 slices plus volumes,
+            or None (default).
+        cmap : str | instance of cmap, optional
+            String or cmap instance specifying colormap.
+        pcnt_range : array-like, optional
+            Percentile range over which to scale image for display.
+        figsize : tuple
+            Figure size (in inches) to use if axes are None.
+        """
+        # Nest imports so that matplotlib.use() has the appropriate
+        # effect in testing
+        plt, _, _ = optional_package('matplotlib.pyplot')
```

Comment:
Well there is a risk if I run tests on my system with e.g. `nosetests nibabel/tests/test_viewers.py` I could get windows that pop up if `show` is not taken care of carefully, or at the very least get some behavior that's different from Travis testing. For now to me it's worth having 3 repeated `plt, _, _, ...` lines to avoid the complications, but if the plotting functions are expected to expand, it might be worth figuring out a better long-term solution.